frontend/apps/reader/modules/readersearch.lua

```diff
@@ -71,13 +82,58 @@ function ReaderSearch:onShowFulltextSearchInput()
     self.input_dialog:onShowKeyboard()
 end
 
-function ReaderSearch:onShowSearchDialog(text, direction)
+function ReaderSearch:onShowRegexSearchInput()
+    local backward_text = "◁"
+    local forward_text = "▷"
+    if BD.mirroredUILayout() then
+        backward_text, forward_text = forward_text, backward_text
+    end
+
+    self.input_dialog = InputDialog:new{
+        title = _("Enter regular expression to search for"),
+        description = _("Don't forget: If your search includes punctuation marks, escape them with '\\'."),
```

Comment:
`.?()[]` only, `,;:!` are fine. I guess if there's room, we'll need a small primer on common char patterns that needs escape and their meaning without escape.

Reply:
I put this message just as a remainder in there, that here we use regex. I would not talk too much about regular expressions here. Those who are familiar with them shall/can use it; other users have to learn it.

So maybe a statement like: "Use ECMAScript regular expression syntax here." might be enough.